File: python/08_002.py
def find_price_pair(prices, budget):
  price_map = {}
  
  for i, price in enumerate(prices):
     complement = budget - price
     
     if complement in price_map:
        return [price_map[complement], i]
     
     price_map[price] = i
  
  return []

File: data structures and algorithms/cs/08_002.cs
using System.Collections.Generic;

public int[] FindPricePair(int[] prices, int budget) {
    Dictionary<int, int> priceMap = new Dictionary<int, int>();
    
    for (int i = 0; i < prices.Length; i++) {
        int complement = budget - prices[i];
        
        if (priceMap.ContainsKey(complement)) {
            return new int[] {priceMap[complement], i};
        }
        
        priceMap[prices[i]] = i;
    }
    
    return new int[] {};  // No solution found
}